src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests_ListPatterns.cs

```diff
@@ -1403,17 +669,14 @@ public void SlicePattern_16()
                         N(SyntaxKind.IdentifierToken, "c");
                     }
                     N(SyntaxKind.IsKeyword);
-                    N(SyntaxKind.RecursivePattern);
+                    N(SyntaxKind.ListPattern);
```

Comment:
nit: consider adding some error cases for `is {} []` or `is () []` (list-pattern is not part of recurisve-pattern)